metals/src/main/scala/scala/meta/internal/metals/Subject.scala

```diff
@@ -0,0 +1,7 @@
+package scala.meta.internal.metals
+
+trait Subject[S] {
```

Comment:
Is `Subject` a common name for this? I normally think of this as an `Observable`?

Reply:
I think it heavily depends on language and people. My first encounter with reactive programming was rxjs and this feature seems like a https://rxjs.dev/guide/subject for me. 
I think the correct hierarchy looks like this
```scala
trait Observable[T] {
  def subscribe(fn: T => Unit): Unit
}

trait Subject[T] extends Observable[T] {
  def push(value: T): Unit
}
```

Reply:
Ah ok, cool. Thanks for the explanation.

Reply:
I'll check Monix and ZIO to see how they name such things. It's better to stick with Scala conventions.